src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -143,7 +166,7 @@ private ReturnTypeChain BuildReturnTypes()
 
         private ConvenienceMethod BuildConvenienceMethod(ReturnTypeChain returnTypeChain)
         {
-            string name = _protocolBodyParameter == null
+            string name = !_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).ToList().Exists(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
                 ? _restClientMethod.Name.IsLastWordSingular()
                     ? $"{_restClientMethod.Name}Value"
                     : $"{_restClientMethod.Name.LastWordToSingular()}Values"
```

Comment:
And BTW this name has two layer of ternary condition operator which is not very readable, please also consider convert this to if else statement in a method and  reduce one layer of the ternary condition